### PR TITLE
FIXES PLT-6648 Add support for Server Side Encryption on S3

### PR DESCRIPTION
**Pull request by tejaycar**
#### Summary
Adds support for AWS S3 server side encryption

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6648

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] All new/modified APIs include changes to the drivers
- [ ] Has enterprise changes (please link)
- [ X] Has UI changes
- [ X] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by coreyhulen** [comment]
You can see the build errors at https://build.mattermost.com/job/mattermost-platform-pr/6804/console

You can run `make test` locally to test your changes.

Checkout https://docs.mattermost.com/developer/dev-setup.html for some helpful tips.

File: config/config.json [hunk 1]
```diff
@@ -123,7 +115,8 @@
         "AmazonS3Bucket": "",
         "AmazonS3Region": "us-east-1",
         "AmazonS3Endpoint": "s3.amazonaws.com",
-        "AmazonS3SSL": true
+        "AmazonS3SSL": true,
+        "AmazonS3SSE": false
```
Comment (coreyhulen) [inline config/config.json]: config.json should only have your 1 change.  Everything else should be changed.

**Comment by jwilander** [comment]
For the most part looks good, just need to fix some syntax errors and get the unit tests running. Here are the syntax errors that Jenkins is reporting:
```
app/file.go:106: copyConds.conditions undefined (cannot refer to unexported field or method conditions)
app/file.go:106: undefined: c in c.conditions
app/file.go:106: undefined: copyCondition
app/file.go:145: S3Metadata(encrypt, "binary/octet-stream") used as value
app/file.go:147: S3Metadata(encrypt, model.GetImageMimeType(ext)) used as value
app/file.go:607: too many arguments to return
	have (map[string][]string)
	want ()
```

**Comment by tejaycar** [comment]
So, I'm running into two problems.

First, minio doesn't support arbitrary metadata on their copy command.  So we can't add SSE to the `moveFile` method until they add that.

Second, I can't seem to get the test to run locally.  When I run `make test` I get this:
```/var/folders/n3/nh9yzqsn0pz48xqztp_2fxnm0000gn/T/go-build813474264/github.com/platform/app/_obj/command.go:184: cannot use args.T (type "github.com/mattermost/platform/vendor/github.com/nicksnyder/go-i18n/i18n".TranslateFunc) as type "github.com/platform/vendor/github.com/nicksnyder/go-i18n/i18n".TranslateFunc in argument to provider.GetCommand
/var/folders/n3/nh9yzqsn0pz48xqztp_2fxnm0000gn/T/go-build813474264/github.com/platform/app/_obj/command.go:389: cannot use utils.T (type "github.com/mattermost/platform/vendor/github.com/nicksnyder/go-i18n/i18n".TranslateFunc) as type "github.com/platform/vendor/github.com/nicksnyder/go-i18n/i18n".TranslateFunc in argument to builtInProvider.GetCommand
/var/folders/n3/nh9yzqsn0pz48xqztp_2fxnm0000gn/T/go-build813474264/github.com/platform/app/_obj/diagnostics.go:279: utils.Cfg.FileSettings.AmazonS3SSE undefined (type model.FileSettings has no field or method AmazonS3SSE)```

But I am declaring the `AmazonS3SSE` and jenkins is not getting the same error.  I'm a total newbee to go, so I'm sure it's just something I'm doing wrong, but I no idea what.

**Comment by tejaycar** [comment]
@jwilander  so we're passing now, but still need the move code which is dependent on https://github.com/minio/minio-go/issues/690

**Comment by tejaycar** [comment]
Needs Minio-go 3.0 before it will pass tests.

**Comment by coreyhulen** [comment]
@jasonblais can you test the system console UI and approve the strings?

**Comment by jasonblais** [comment]
@tejaycar 

Does this PR add AWS Server-Side Encryption for files in Amazon S3, using Amazon S3-Managed Keys http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+* -text
```
Comment (coreyhulen) [inline .gitattributes]: Can we remove this file?

**Comment by tejaycar** [comment]
@jasonblais That is correct. This uses S3-managed keys.  Using customer managed keys, or even IAM keys, would be significantly more complicated and would be better suited for EE (though either or both would be great in EE)

**Comment by tejaycar** [comment]
@coreyhulen sorry about that.  I had trouble getting my line endings to behave and accidentally checked in the file that fixed it for me.

**Comment by jasonblais** [comment]
Okay cool, thanks @tejaycar  

I'm just getting the text approved, will share shortly if there are any tweaks.

**Comment by jasonblais** [comment]
@tejaycar I tried submitting a PR against your branch but it gave me an error, so I submitted small changes to the System Console UI text directly.

Hope that's okay? Big thanks for your help with this.

@coreyhulen FYI UI text reviewed and approved by Ian.

**Comment by tejaycar** [comment]
@jasonblais  yea, totally ok.  I think I even had a check box when I created the PR that gave you permission to push changes to it.  Much easier than asking me to make small changes that you need so you can accept my changes :).  That starts to feel a little too meta ;)

**Comment by coreyhulen** [comment]
@tejaycar looks like you might have a build error

```
github.com/mattermost/platform/model.BuildEnterpriseReady=true" ./cmd/platform
# github.com/mattermost/platform/app
app/file.go:651: undefined: metaData
app/file.go:652: undefined: metaData
app/file.go:653: undefined: metaData
make: *** [build-linux] Error 2
```

**Comment by coreyhulen** [comment]
@tejaycar you have an error in some formatting

```
go vet  ./model || exit 1
vet: model/config_test.go: model/config_test.go:26:6: expected '(', found 'IDENT' TestConfigDefaultFileSettingsS3SSE (and 1 more errors)
vet: no files checked
exit status 1
```

**Pull request merged**